File: src/main/java/com/dude/dms/ui/builder/attributes/AttributeValueFieldBuilder.java
package com.dude.dms.ui.builder.attributes;

import com.dude.dms.backend.data.docs.AttributeValue;
import com.dude.dms.backend.service.AttributeValueService;
import com.dude.dms.ui.components.tags.AttributeValueField;

public final class AttributeValueFieldBuilder {

    private final AttributeValue attributeValue;

    private final AttributeValueService attributeValueService;

    AttributeValueFieldBuilder(AttributeValue attributeValue, AttributeValueService attributeValueService) {
        this.attributeValue = attributeValue;
        this.attributeValueService = attributeValueService;
    }

    public AttributeValueField build() {
        return new AttributeValueField(attributeValue, attributeValueService);
    }
}

File: src/main/java/com/dude/dms/ui/builder/attributes/AttributeValueFieldBuilder.kt
package com.dude.dms.ui.builder.attributes

import com.dude.dms.backend.data.docs.AttributeValue
import com.dude.dms.backend.service.AttributeValueService
import com.dude.dms.ui.builder.Builder
import com.dude.dms.ui.components.tags.AttributeValueField

class AttributeValueFieldBuilder(
        private val attributeValue: AttributeValue,
        private val attributeValueService: AttributeValueService
): Builder<AttributeValueField> {

    override fun build() = AttributeValueField(attributeValue, attributeValueService)
}